### ISSUE TITLE
Plugins: Incompatible plugins need to be disabled for AT sites

### ISSUE THREAD

**Issue by rralian** [Automattic/wp-calypso#12131]
Some plugins are incompatible with the AT site structure. For these plugins, we should still show the plugin details page and let them appear in search results. But we should replace the "install" functionality with a notice that the plugin is not supported.

- nginx-helper
- w3-total-cache
- wp-rocket
- wp-super-cache
- bwp-minify

For now I would say just remove the "install" button and show a message on the page saying "This plugin is not supported on WordPress.com".

/cc @roundhill @drw158

**Comment by drw158** [comment]
My small suggestion would be to disable the install button, and display a notice at the top. Probably a yellow notice.

**Comment by kriskarkoski** [comment]
I can put together a support page for these

**Comment by kriskarkoski** [comment]
We should make sure they only show up in direct searches and are no longer suggested once this is addressed. wp-super-cache in particular seems to appear in the Featured section frequently

**Comment by vindl** [comment]
@kriskarkoski Could you send me a link to the support page so I can include it in the PR? (doesn't need to be finished).

**Comment by kriskarkoski** [comment]
@vindl I've published http://en.support.wordpress.com/incompatible-plugins/ (currently empty) so you can use that link. @rudzki is going to work on the content tomorrow.

**Comment by folletto** [comment]
> "This plugin is not supported on WordPress.com" Sounds good to me. It would be worth creating a .com support page and link to it, explaining why they are not supported

FWIW: are these plugins "just" not supported, or our infrastructure already provides most of the benefits they'll provide too? 

In the first case I think the text is good to go. If they aren't supported AND we already provide that functionality with other means, I think we should say that?

So IF that is the case, maybe something like:

> Incompatible plugin: WordPress.com already provides the benefits of this plugin.

Or maybe keep the message and just write that on the Support page?

**Comment by drw158** [comment]
https://kb.pressable.com/faq/does-pressable-restrict-any-plugins/

I know that all of them except for bwp-minify provide redundant caching. bwp-minify actually slows down the site.

I think I would shorten it just a bit:

> Incompatible plugin: WordPress.com already provides this feature.

**Comment by drw158** [comment]
For reference, here is the issue that tried to deal with plugins that enable features we already have #11064

We came to the conclusion that unless there is a **real conflict** (like the plugins here is this issue) then we shouldn't display a warning on the plugins page.

**Comment by rudzki** [comment]
@kriskarkoski https://en.support.wordpress.com/incompatible-plugins/ seems sufficient for now, unless you have suggestions?

**Comment by GeekPress** [comment]
Hi,

Jonathan from WP Rocket.

Is it possible to know what is the issue with the AT site structure?

Maybe we could do something our side to be compatible :)

Kind Regards

**Comment by kriskarkoski** [comment]
@BandonRandon I think you might be able to help with this question ^

**Comment by BandonRandon** [comment]
Hi @GeekPress!

As the AT site structure is a managed WordPress host we've included a managed caching system. We use a combination of WP Super Cache and Batcache on the infrastructure. As I'm sure you know caching plugins don't always play nice together. Unfortunately this extends to our implementation of WP Super Cache and WP-Rocket.

Part of this managed infrastructure is managing the `advanced-cache.php` file for our customers. When other caching plugins, like WP-Rocket try to add,modify, or delete the `advanced-cache.php` file they will be greeted with a permission denied error.

I've just submitted your contact form requesting a copy of the plugin to take a deeper dive into your code.
